sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -40,16 +39,38 @@ private[execution] sealed trait HashedRelation {
   /**
     * Returns matched rows.
     */
-  def get(key: InternalRow): Seq[InternalRow]
+  def get(key: InternalRow): Iterator[InternalRow]
```

Comment:
we should make it clear in the documentation what this is supposed to return when there is no match (null or empty iterator). it actually seems to me empty iterator is safer?